File: src/org/jetbrains/jet/j2k/ast/PolyadicExpression.java
package org.jetbrains.jet.j2k.ast;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.jet.j2k.util.AstUtil;

import java.util.List;

/**
 * @author ignatov
 */
public class PolyadicExpression extends Expression {
  private final List<Expression> myExpressions;
  private final String myToken;
  private final List<String> myConversions;

  public PolyadicExpression(List<Expression> expressions, String token, List<String> conversions) {
    super();
    myExpressions = expressions;
    myToken = token;
    myConversions = conversions;
  }

  @NotNull
  @Override
  public String toKotlin() {
    List<String> expressionsWithConversions = AstUtil.applyConversions(AstUtil.nodesToKotlin(myExpressions), myConversions);
    return "(" + AstUtil.join(expressionsWithConversions, SPACE + myToken + SPACE) + ")";
  }
}

File: src/org/jetbrains/jet/j2k/ast/PolyadicExpression.kt
package org.jetbrains.jet.j2k.ast


public open class PolyadicExpression(val expressions: List<Expression>, val token: String): Expression() {
    public override fun toKotlin(): String {
        val expressionsWithConversions = expressions.map { it.toKotlin() }
        return expressionsWithConversions.makeString(" " + token + " ")
    }
}
